addons/website_sale_referral/models/referral_mixin.py

```diff
@@ -0,0 +1,82 @@
+from odoo import api, fields, models, _, SUPERUSER_ID
+
+
+class MailActivity(models.Model):
+    _inherit = 'mail.activity'
+
+    def _action_done(self, feedback=False, attachment_ids=None):
+        obj = self.env[self.res_model].browse(self.res_id)
+        obj.reward_done = True
+        return super(MailActivity, self)._action_done(feedback=feedback, attachment_ids=attachment_ids)
+
+
+class ReferralMixin(models.AbstractModel):
+    """ Mixin class for objects which can be tracked by referral. """
+    _name = 'referral.mixin'
+    _description = 'Referral Mixin'
+
+    STATES_PRIORITY = {'cancel': 0, 'new': 1, 'in_progress': 2, 'done': 3}
+    REFERRAL_STAGES = {'new': 'New', 'in_progress': 'In Progress', 'done': 'Done'}
+
+    to_reward = fields.Boolean()
+    reward_done = fields.Boolean()
```

Comment:
It was designed to not have to make that check. Reward_done encompass to_reward.
- If the code change in the future, and somebody forgets to change both fields, it could mess everything up.
- Allows to avoid checking both fields in many places.

I agree that this wasn't clear, I have renamed to_reward to deserve_reward to make it more clear.

Reply:
And I added a description